From: SeongJae Park <sj@kernel.org>
Date: Sat, 28 Jun 2025 14:30:56 -0700
Subject: [PATCH] ==== damon_callback deprecation ====

mm/damon: remove damon_callback

damon_callback was the only way for communicating with DAMON for
contexts that running on its worker thread.  The interface is flexible
and simple.  But as DAMON evolves with more features, damon_callback
became somewhat too old.  With runtime parameters update, for example,
its lack of synchronization support was found to be inconvenient.
Arguably it is also not easy to use correctly since the callers should
understand when each callback is called, and implication of the return
values from the callbacks.

To replace it, damon_call() and damos_walk() are introduced and replaced
multiple damon_callback use case.  Some use cases of damon_callback such
as parallel or repetitive DAMON internal data reading and additional
cleanups cannot simply replaced by damon_call() and damos_walk(),
though.

To allow those replaceable, extend damon_call() for parallel and/or
repeated callbacks and modify the core/ops layers for additional
resources cleanup.  With the updates, replace the remaining
damon_callback usages and finally remove damon_callback.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_d7y45blw | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_d7y45blw

diff --git a/hkml_cv_bogus/hkml_cv_bogus_d7y45blw b/hkml_cv_bogus/hkml_cv_bogus_d7y45blw
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

